src/blocks/blocks/form/edit.js

```diff
@@ -200,191 +610,221 @@ const Edit = ({
 				}
 			});
 		}
-	}, [ attributes.hasCaptcha, settingsRef.current ]);
+		return () => controller?.abort();
+	}, [ attributes.hasCaptcha, attributes.optionName ]);
 
 	/**
-	 * Check if the API Keys are set.
+	 * Check if the reCaptcha API Keys are set.
 	 */
 	useEffect( () => {
-		let isMounted = true;
-
-		const getAPIData = async() => {
-			if ( ! isAPILoaded ) {
-				settingsRef?.current?.fetch().then( response => {
-					setAPILoaded( true );
+		let controller = new AbortController();
+		const getCaptchaAPIData = () => {
+			setLoading({ captcha: 'loading'});
+			( new api.models.Settings() )?.fetch({ signal: controller.signal }).then( response => {
+				controller = null;
+
+				if ( '' !== response.themeisle_google_captcha_api_site_key && '' !== response.themeisle_google_captcha_api_secret_key ) {
+					setLoading({ captcha: 'done'});
+				} else {
+					setLoading({ captcha: 'missing'});
+					setGoogleCaptchaAPISiteKey( response.themeisle_google_captcha_api_site_key );
+					setGoogleCaptchaAPISecretKey( response.themeisle_google_captcha_api_secret_key );
+				}
+			}).catch( e => {
+				console.error( e );
+				setLoading({ captcha: 'error' });
+			});
 
-					if ( '' !== response.themeisle_google_captcha_api_site_key && '' !== response.themeisle_google_captcha_api_secret_key && isMounted ) {
-						setAPISaved( true );
-					}
-				});
-			}
 		};
 
-		if ( areSettingsAvailable && attributes.hasCaptcha && ! isAPISaved ) {
-			getAPIData();
+		if ( attributes.hasCaptcha && 'init' === loadingState?.captcha ) {
+			getCaptchaAPIData();
 		}
 
-		return () => {
-			isMounted = false;
-		};
-	}, [ areSettingsAvailable, isAPILoaded, isAPISaved, attributes.hasCaptcha ]);
+		return () => controller?.abort();
+	}, [ loadingState.captcha, attributes.hasCaptcha ]);
 
 	/**
 	 * Save API Keys in the Otter options.
 	 */
-	const saveAPIKey = () => {
-		const model = new window.wp.api.models.Settings({
+	const saveCaptchaAPIKey = () => {
+		setLoading({ captcha: 'loading' });
+		const model = new api.models.Settings({
```

Comment:
From next time, not right now, you can maybe start using useSettings hook to do it easily: https://gist.github.com/HardeepAsrani/950f4796759a5c7a29fb537837e3a47d